Sources/Common/Concurrency/TaskExtension.swift

```diff
@@ -20,13 +20,12 @@ import Foundation
 
 public extension Task where Success == Never, Failure == Error {
 
-    static func periodic(priority: TaskPriority? = nil,
-                         delay: TimeInterval? = nil,
+    static func periodic(delay: TimeInterval? = nil,
                          interval: TimeInterval,
                          operation: @escaping @Sendable () async -> Void,
                          cancellationHandler: (@Sendable () async -> Void)? = nil) -> Task {
 
-        Task.detached(priority: priority) {
```

Comment:
Detached tasks need to be cancelled explicitly.  We weren't doing that anywhere.

IMHO that's likely to happen again because this `periodic` method abstracts the caller from knowing there's a detached task in here.  Since this method is only used by us right now, and it works well with a regular task, I'm just changing it to use that, instead.